GoogleForCreators/web-stories-wp
# New Story Details Modal: Inspector Context and Panel Adjustments

## Context

We need the document panel in the new publish modal. 

## Summary

Moves the Inspector up in hierarchy to access from header, wires up Document Pane for modal, some style updates.


## Relevant Technical Choices

Inspector Context
- Moving the Inspector Provider up from `Inspector` to [`Layout`](https://github.com/GoogleForCreators/web-stories-wp/pull/10518/files#diff-e07426bc8115232e1efac75b9510d316cfe08e0ec5441725a361446a9badbd32R90) so that we can access the `DocumentPane` in the new Story Details modal (via header publish button).

`Panel` Updates: 
- [Updating the `Title`](https://github.com/GoogleForCreators/web-stories-wp/pull/10518/files#diff-cb4fb18e75e611c59228354d45b53a41ec97a7a885dba78d148a92d8bebe5f58L257) to conditionally render the `IconWrapper` associated with the toggle to collapse because when you can't collapse there's an extra space where the icon should be and when you hover over it there's a gray circle? I'm pretty sure this is the first case of an actually non-collapsing panel but I want to call it out incase this causes a regression I'm unaware of.  
- Passing `props` through on all inspector panels so that we can override `Panel` props - right now I need `name`, `canCollapse` and `isPersistable` - being able to override them is important to preventing weird cross over "persistence" from opened panels in the editor vs modal - they need to persist independently.
- Pulling all of the panel names into constants so that a list of them can be assembled reliably, we need this to conditionally render them in the modal vs all of them in the editor - it feels bad typing loose strings for these so I tied them together. Only exception is the `status` panel which lives in wp-story-editor and i'm not sure sure what to do with it, there's a [note here](https://github.com/GoogleForCreators/web-stories-wp/pull/10518/files#diff-23115102a5db709933b9ff35192fc363c3cbfb70cc9eac1ac7d53a9cc94162d3R72).

wp-story-editor Updates: 
- Added two new exports to `/documentPanel` - `PublishModalDocumentPane` and `IsolatedStatusPanel` 
- Initially I had updated the `DocumentPane` to pass in some props and then decide dynamically what panels to render, but I think the DocumentPanel should be kept as is and used accordingly. Max had a good idea to simplify it. 
- So now in the `inspectorTab` that's getting set up in `wp-story-editor`'s `Layout` and [sent in as a new object to `story-editor` as `publishModal` where those two new components from documentPane are getting added as values](https://github.com/GoogleForCreators/web-stories-wp/pull/10518/files#diff-638798d8686c20fab74b41ca59460253ed4f7e5c796923421e722bb49d06b29aR56). 



## To-do
- [x] Tests updated accordingly 
- [x] See idea through for conditionally rendering 1 pane 
- [x] Fix horizontal styling (scroll) 
- [x] RTL styling fixed 
- [x] New Karma tests 

Separately: 
- Rest of checklist here: https://github.com/GoogleForCreators/web-stories-wp/issues/10115 
- e2e tests to cover functionality of document panel that is WordPress specific
- When publish date moves to the future the copy on the "publish" button should be "Schedule"
- Need to update the visibility logic so that when you private publish immediately it doesn't ask about the checklist and loop you back to the story details modal 😂 
- Also, the `Visibility` panel needs an update

| From this | to this | 
| --- | --- | 
| <img width="289" alt="Screen Shot 2022-02-10 at 2 33 26 PM" src="https://user-images.githubusercontent.com/10720454/153500174-c0e4b918-ad9c-4234-8cf1-ae94f1458856.png"> | <img width="195" alt="Screen Shot 2022-02-10 at 2 33 46 PM" src="https://user-images.githubusercontent.com/10720454/153500221-8edd6b50-1233-4fa7-8167-ad5cdbd4ec56.png"> | 

## User-facing changes

(Behind feature flag still, when flag is enabled this is what's new) 
| LTR | RTL | 
| --- | --- |
| ![ltr panel demo](https://user-images.githubusercontent.com/10720454/153497019-280f1d9f-4737-46b2-8420-44759193ac86.gif) | ![rtl ](https://user-images.githubusercontent.com/10720454/153497038-bdb444d4-6728-44ef-abbc-eef9b8f89c33.gif) | 

Storybook can demo the panels too - just a very dumb downed version to avoid having to feed a bunch of providers to it.
<img width="1113" alt="Screen Shot 2022-02-10 at 2 09 50 PM" src="https://user-images.githubusercontent.com/10720454/153497149-25568f7d-0680-4733-9e6d-380bc05886d6.png">



## Testing Instructions

- Yup you guessed it! More regression testing! This time it's the document panel that needs it. Everything should work exactly the same as in staging. 
- Now, with `Updated Publish Story Modal` feature flag enabled, hit 'publish' and see the document panel present in the modal. It has some differences compared to the document panel in the editor. I will outline them below, you can see[ figma ](https://www.figma.com/file/ZrO763yvMWJjYeAPsRnNfW/_Editor---Sprint-Feature-Level-Designs_2022?node-id=57%3A14792)for designs.  
  - Modal has: `Publishing`, `Permalink`, `Page Advancement`, `Background Audio`, `Taxonomies` (5) panels 
  - Modal has isolated `Visibility` "panel" below the `Story Description` in the middle section  
  - If you are looking at designs in figma and notice that the visibility panel looks different, yup. I'm just using the existing design. We can update what it looks like separately.
- Open panel persistence should NOT cross between modal and editor. That is to say, if you have all of your document pane panels open in the editor and then open the modal and collapse 'Publishing' and 'Permalink' and then close the modal, the editor's version of those two panels should still be open. 






This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Partially addresses #10115
